src/qibocal/protocols/characterization/state_tomography.py

```diff
@@ -0,0 +1,285 @@
+
+from .utils import table_dict, table_html
+
+
+@dataclass
+class StateTomographyParameters(Parameters):
+    """Tomography input parameters"""
+
+    circuit: Optional[Union[str, Circuit]] = None
+    """Circuit to prepare initial state.
+
+        It can also be provided the path to a json file containing
+        a serialized circuit.
+    """
+
+    def __post_init__(self):
+        if isinstance(self.circuit, str):
+            raw = json.loads((Path.cwd() / self.circuit).read_text())
+            self.circuit = Circuit.from_dict(raw)
+
+
+TomographyType = np.dtype(
+    [
+        ("samples", np.int64),
+    ]
+)
+"""Custom dtype for tomography."""
+
+
+@dataclass
+class StateTomographyData(Data):
+    """Tomography data"""
+
+    x_basis_state: QuantumState = None
+    y_basis_state: QuantumState = None
+    z_basis_state: QuantumState = None
+    data: dict[tuple[QubitId, str], np.float64] = field(default_factory=dict)
+
+    def save(self, path):
+        self.x_basis_state.dump(path / "x.json")
+        self.y_basis_state.dump(path / "y.json")
+        self.z_basis_state.dump(path / "z.json")
+        self._to_npz(path, DATAFILE)
+
+    @classmethod
+    def load(cls, path):
+        instance = cls()
+        instance.data = super().load_data(path, DATAFILE)
+        instance.x_basis_state = QuantumState.load(path / "x.json")
+        instance.y_basis_state = QuantumState.load(path / "y.json")
+        instance.z_basis_state = QuantumState.load(path / "z.json")
+
+        return instance
+
+
+@dataclass
+class StateTomographyResults(Results):
+    """Tomography results"""
+
+    density_matrix_real: dict[QubitId, list]
+    """Real part of measured density matrix."""
+    density_matrix_imag: dict[QubitId, list]
+    """Imaginary part of measured density matrix."""
+    target_density_matrix_real: list
+    """Real part of exact density matrix."""
+    target_density_matrix_imag: list
+    """Imaginary part of exact density matrix."""
+    fidelity: dict[QubitId, float]
+    """State fidelity."""
+
+
+def _acquisition(
+    params: StateTomographyParameters, platform: Platform, targets: list[QubitId]
+) -> StateTomographyData:
+    """Acquisition protocol for single qubit state tomography experiment."""
+
+    if params.circuit is None:
+        params.circuit = Circuit(len(targets))
+        params.circuit.add(gates.H(i) for i in range(len(targets)))
+
+    backend = GlobalBackend()
+    transpiler = dummy_transpiler(backend)
+
+    data = StateTomographyData()
+
+    for basis in ["X", "Y", "Z"]:
+        basis_circuit = deepcopy(params.circuit)
+        # FIXME: basis
+        if basis != "Z":
+            for i in range(len(targets)):
+                basis_circuit.add(getattr(gates, basis)(i).basis_rotation())
+
+        # basis_circuit.add(gates.M(i, basis=getattr(gates, basis)) for i in targets)
+        for i in range(len(targets)):
+            basis_circuit.add(gates.M(i))
```

Comment:
```suggestion
        basis_circuit.add(gates.M(i) for i in range(len(targets)))
```
should still work (following the comment), but regardless I would still drop the comment (dead code).

Reply:
This is still related to https://github.com/qiboteam/qibo/issues/1318, I would drop the dead code and provide a reference to the issue.

Reply:
Actually no, my bad... I can implement it